aspnetcore/blazor/lifecycle.md

```diff
@@ -63,7 +63,12 @@ If `base.SetParametersAync` isn't invoked, the custom code can interpret the inc
 
 ### After parameters are set
 
-<xref:Microsoft.AspNetCore.Components.ComponentBase.OnParametersSetAsync*> and <xref:Microsoft.AspNetCore.Components.ComponentBase.OnParametersSet*> are called when a component has received parameters from its parent and the values are assigned to properties. These methods are executed after component initialization and each time new parameter values are specified:
+<xref:Microsoft.AspNetCore.Components.ComponentBase.OnParametersSetAsync*> and <xref:Microsoft.AspNetCore.Components.ComponentBase.OnParametersSet*> are called:
+
+* After component initialization when its parent component is rendered.
+* When a component has assigned values to properties:
+  * From *changes* to simple parameters by its parent.
+  * When complex-type parameters *are used*, even when complex-type parameter values don't change.
```

Comment:
For this section, I find the phrasing a bit confusing. I'd say something like:

`OnParametersSetAsync` and `OnParametersSet` are called:

 * When the component is being initialized and has received its first set of parameters from its parent component
 * When the parent component re-renders and supplies a possibly-changed set of parameters
    * As a special case, if the parameters are all of known primitive immutable types, the framework will skip re-rendering the child if their values are unchanged. 
    * If there are any complex-typed parameters, the framework cannot know whether the values have mutated internally so it must always treat the parameter set as changed.

Reply:
> - When the component is being initialized and has received its first set of parameters from its parent component

Could you tweak this a bit to indicate they are called after OnInitialized?

Reply:
I'll remove "being," so it will read ...

> When a component is initialized and ...

These sections are in order. I *think* it will be clear. We'll see. We'll get feedback :ear: if there's confusion on the order.